src/storage/rdb.cc

```diff
@@ -156,10 +173,9 @@ StatusOr<std::string> RDB::loadEncodedString() {
   }
 
   // Normal string
-  GET_OR_RET(peekOk(static_cast<size_t>(len)));
-  auto value = std::string(input_.data() + pos_, len);
-  pos_ += len;
-  return value;
+  std::vector<char> vec(len);
+  GET_OR_RET(stream_->Read(vec.data(), len));
```

Comment:
Could it be `len == 0`?

Reply:
Yes, it is possible, could you  help me fix it?

Reply:
https://github.com/apache/kvrocks/pull/1839
I've create a pull request for it.

Thanks again for your contribution!